vspace/src/main/webapp/WEB-INF/views/staff/spaces/space.html

```diff
@@ -24,309 +27,339 @@
-	$("#spaceLinkInfoLabel").text(spaceLinkName);
-	$("#spaceLinkDisplayId").val(displayLinkId);
-	$("#spaceLinkInfoLabelEdit").text(spaceLinkName);
-	$("#spaceLinkLabelEdit").val(spaceLinkName);
-	$("#spaceLinkId").val(spaceLinkId);
-	$("#spaceLinkIdValueEdit").val(spaceLinkId);
-	$("#spaceLinkRotationEdit").val(rotation);
-	$('#typeSpaceEdit option[value="'+linkType+'"]').attr("selected", "selected");
-	$('#spaceLinkIdEdit option[value="'+selectedSpaceId+'"]').attr("selected", "selected");
-    resetHighlighting(); 
-    $('[data-link-id="' + spaceLinkId + '"]').css("color", "#c1bb88");
-    $('div[data-link-id="' + spaceLinkId + '"]').removeClass("alert-primary");
-    $('div[data-link-id="' + spaceLinkId + '"]').addClass("alert-warning");
-    $('img[data-link-id="' + spaceLinkId + '"]').css("border", "solid 1px #c1bb88");
-    $('[data-link-id="unpublished-' + spaceLinkId + '"]').css("color", "#c1bb88");
-    
-    $("#bgImage").on("click", function(e){
-		e.preventDefault();
-		var posX = $(this).position().left;
-		var posY = $(this).position().top;    
-		storeX = e.pageX - $(this).offset().left;
-		storeY = e.pageY - $(this).offset().top;
-		showSpaceLinkEdit(editSpaceLinkInfo(),posX,posY);
-	});
-    hideLinkInfoTabs();
-    $("#editSpaceLinkInfo").show();
+//link icons for space links
+var linkIconReader = new FileReader();
+var linkIcon;
+linkIconReader.onload = function(e) {
+	linkIcon = e.target.result;
+	showSpaceLink(createSpaceLinkInfo());
 }
 
-function makeModuleLinksEditable(moduleLinkName, moduleLinkId, rotation, selectedModuleId, posXEdit, posYEdit, displayLinkId, linkType) {
-    
-    selectedModuleLinkId=moduleLinkId;
-    storeX=posXEdit;
-	storeY=posYEdit;
-	$("#moduleLinkInfoLabel").text(moduleLinkName);
-	$("#moduleLinkDisplayId").val(displayLinkId);
-	$("#moduleLinkInfoLabelEdit").text(moduleLinkName);
-	$("#moduleLinkLabelEdit").val(moduleLinkName);
-	$("#moduleLinkId").val(moduleLinkId);
-	$("#moduleLinkIdValueEdit").val(moduleLinkId);
-	$("#moduleLinkRotationEdit").val(rotation);
-	$('#typeEdit option[value="'+linkType+'"]').attr("selected", "selected");
-	$('#moduleLinkIdEdit option[value="'+selectedModuleId+'"]').attr("selected", "selected");
-	resetHighlighting();     
-	$('[data-link-id="' + moduleLinkId + '"]').css("color", "#c1bb88");
-	$('div[data-link-id="' + moduleLinkId + '"]').removeClass("alert-primary");
-	$('div[data-link-id="' + moduleLinkId + '"]').addClass("alert-warning");
-	$('img[data-link-id="' + moduleLinkId + '"]').css("border", "solid 1px #c1bb88");
-	
-	$("#bgImage").on("click", function(e){
-		e.preventDefault();
-		var posX = $(this).position().left;
-		var posY = $(this).position().top;    
-		storeX = e.pageX - $(this).offset().left;
-		storeY = e.pageY - $(this).offset().top;
-		showModuleLinkEdit(editModuleLinkInfo());
-	});
-	hideLinkInfoTabs();
-    $("#editModuleLinkInfo").show();
+//link icons for module links
+var moduleLinkIconReader = new FileReader();
+var moduleLinkIcon;

```

Comment:
is this variable used outside of the onload function? if not, declare where needed.

Reply:
It is being used outside the onload function for the links of type image.

Reply:
let's pass that into the function (it's only used with `showModuleLink`), right?

Reply:
We don't really have an Image type for Module links. Should I remove this completely?

Reply:
If it's not used, then yes, it shouldn't be there.

Reply:
Shouldn't that be removed per above discussion?